app/controllers/holds_controller.rb

```diff
@@ -52,21 +52,27 @@ def cancel
   end
 
   def create
-    set = TeacherSet.find(params[:teacher_set_id])
-    @hold = set.holds.build(params[:hold])
-    @hold.user = current_user
-    
-    unless params[:settings].nil?
-      current_user.update_attributes(params[:settings])
-    end
+    begin
+      set = TeacherSet.find(params[:teacher_set_id])
+      @hold = set.holds.build(params[:hold])
+      @hold.user = current_user
 
-    respond_to do |format|
-      if @hold.save
-        format.html { redirect_to hold_url(@hold.access_key), notice: 'Your order has been received by our system and will soon be delivered to your school.<br/><br/>Check your email inbox for a message with further details.' }
-        format.json { render json: @hold, status: :created, location: @hold }
-      else
-        format.html { render action: "new" }
-        format.json { render json: @hold.errors, status: :unprocessable_entity }
+      unless params[:settings].nil?
+        current_user.update_attributes(params[:settings])
+      end
+      
+      respond_to do |format|
+        if @hold.save
+          format.html { redirect_to hold_url(@hold.access_key), notice: 'Your order has been received by our system and will soon be delivered to your school.<br/><br/>Check your email inbox for a message with further details.' }
+          format.json { render json: @hold, status: :created, location: @hold }
+        else
+          format.html { render action: "new" }
```

Comment:
might need to set response.error here, too

Reply:
What is response.error?